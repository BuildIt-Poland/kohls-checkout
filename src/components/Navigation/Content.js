import styled from 'styled-components';

import media from '../../styles/media';
import ContentBase from '../Content';

const Content = styled(ContentBase)`
  display: flex;
  height: 6rem;

  ${media.tablet`
    padding: 0;
  `};
`;

export default Content;
