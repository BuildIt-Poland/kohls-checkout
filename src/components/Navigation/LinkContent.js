import styled from 'styled-components';

import { colorBlack, colorWhite, fontMedium, fontWeightBold, colorLightGrey } from '../../styles/designTokens';

function getFontColor({ isActive, isBlocked }) {
  if (isActive) {
    return colorWhite;
  } else if (isBlocked) {
    return colorLightGrey;
  }

  return colorBlack;
}

const LinkContent = styled.span`
  background: ${({ isActive }) => (isActive ? colorBlack : colorWhite)};
  color: ${getFontColor};
  display: flex;
  font-size: ${fontMedium};
  font-weight: ${fontWeightBold};
  justify-content: center;
  align-items: center;
  width: 100%;
`;

export default LinkContent;
