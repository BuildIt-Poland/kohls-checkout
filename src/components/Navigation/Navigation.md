```js
import { withRouter } from 'react-router-dom';

const RoutedNav = withRouter(Navigation);

<RoutedNav />
```
