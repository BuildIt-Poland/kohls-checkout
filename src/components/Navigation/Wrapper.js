import styled from 'styled-components';

import { colorLightGrey, borderWidthThin } from '../../styles/designTokens';

const Wrapper = styled.div`
  width: 100%;
  border-bottom: ${borderWidthThin} solid ${colorLightGrey};
`;

export default Wrapper;
