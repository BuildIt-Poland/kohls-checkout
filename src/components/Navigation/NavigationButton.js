import React from 'react';
import styled from 'styled-components';
import { Link } from 'react-router-dom';

import { borderWidthThin, colorLightGrey, borderWidthXThick, colorYellow } from '../../styles/designTokens';
import media from '../../styles/media';
import LinkContent from './LinkContent';

const FlexLink = styled(Link)`
  display: flex;
  width: 100%;
  text-decoration: none;
  border-left: ${borderWidthThin} solid ${colorLightGrey};
  pointer-events: ${({ disabled }) => (disabled ? 'none' : 'all')};
  &:last-child {
    border-right: ${borderWidthThin} solid ${colorLightGrey};
  }

  ${media.tablet`
    &:first-child {
      border-left: 0;
    }

    &:last-child {
      border-right: 0;
    }
  `};

  &:focus {
    outline: ${borderWidthXThick} solid ${colorYellow};
    outline-offset: -${borderWidthXThick};
  }
`;

function NavigationButton({ to, isActive, isBlocked, children }) {
  return (
    <FlexLink data-testid="navigation-link" to={to} disabled={isBlocked}>
      <LinkContent isActive={isActive} isBlocked={isBlocked}>
        {children}
      </LinkContent>
    </FlexLink>
  );
}

export default NavigationButton;
