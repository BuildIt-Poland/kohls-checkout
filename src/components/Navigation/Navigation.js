import React from 'react';
import styled from 'styled-components';

import { fontWeightNormal } from '../../styles/designTokens';
import { CHECKOUT_DELIVERY_PATH, CHECKOUT_PAYMENT_PATH, CHECKOUT_REVIEW_PATH } from '../../constants/routes';
import ErrorBoundary from '../ErrorBoundary';
import NavigationButton from './NavigationButton';
import Wrapper from './Wrapper';
import Content from './Content';

const STEPS = {
  DELIVERY: 'delivery',
  PAYMENT: 'payment',
  REVIEW: 'review'
};

const StepNumber = styled.span`
  font-weight: ${fontWeightNormal};
  padding-right: 0.4rem;
`;

function Navigation({ match }) {
  const isStepActive = step => step === match.params.checkoutStep;
  const isStepDelivery = () => match.params.checkoutStep === STEPS.DELIVERY;
  const isStepPayment = () => match.params.checkoutStep === STEPS.PAYMENT;

  return (
    <ErrorBoundary>
      <Wrapper>
        <Content as="nav">
          <NavigationButton to={CHECKOUT_DELIVERY_PATH} isActive={isStepActive(STEPS.DELIVERY)}>
            <StepNumber>1.</StepNumber>Delivery
          </NavigationButton>
          <NavigationButton
            to={CHECKOUT_PAYMENT_PATH}
            isActive={isStepActive(STEPS.PAYMENT)}
            isBlocked={isStepDelivery()}
          >
            <StepNumber>2.</StepNumber>Payment
          </NavigationButton>
          <NavigationButton
            to={CHECKOUT_REVIEW_PATH}
            isActive={isStepActive(STEPS.REVIEW)}
            isBlocked={isStepDelivery() || isStepPayment()}
          >
            <StepNumber>3.</StepNumber>Review
          </NavigationButton>
        </Content>
      </Wrapper>
    </ErrorBoundary>
  );
}

export default Navigation;
