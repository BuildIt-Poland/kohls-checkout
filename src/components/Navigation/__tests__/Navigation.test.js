import React from 'react';
import { create } from 'react-test-renderer';

import Navigation from '../Navigation';

jest.mock('react-router-dom', () => ({
  Link: 'Link'
}));

describe('COMPONENT - Navigation', () => {
  it('renders Navigation component with "Delivery" tab set as Active', () => {
    const match = {
      params: {
        checkoutStep: 'delivery'
      }
    };

    const component = create(<Navigation match={match} />);

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders Navigation component with "Payment" tab set as Active', () => {
    const match = {
      params: {
        checkoutStep: 'payment'
      }
    };

    const component = create(<Navigation match={match} />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders Navigation component with "Review" tab set as Active', () => {
    const match = {
      params: {
        checkoutStep: 'review'
      }
    };

    const component = create(<Navigation match={match} />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
