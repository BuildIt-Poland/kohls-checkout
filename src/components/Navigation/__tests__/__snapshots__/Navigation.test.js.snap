// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT - Navigation renders Navigation component with "Delivery" tab set as Active 1`] = `
.c2 {
  background: #222222;
  border-right: 0;
  border: 1px solid #454545;
  color: #ffffff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1.2rem;
  font-weight: 500;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 15px;
  width: 100%;
}

.c2:last-child {
  border-right: 1px;
}

.c3 {
  background: #ffffff;
  border-right: 0;
  border: 1px solid #454545;
  color: #222222;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1.2rem;
  font-weight: 500;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 15px;
  width: 100%;
}

.c3:last-child {
  border-right: 1px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: fixed;
  width: 100%;
}

<nav
  className="c0"
>
  <Link
    className="c1"
    to="/checkout/delivery"
  >
    <span
      className="c2"
    >
      Delivery
    </span>
  </Link>
  <Link
    className="c1"
    to="/checkout/payment"
  >
    <span
      className="c3"
    >
      Payment
    </span>
  </Link>
  <Link
    className="c1"
    to="/checkout/review"
  >
    <span
      className="c3"
    >
      Review
    </span>
  </Link>
</nav>
`;

exports[`COMPONENT - Navigation renders Navigation component with "Payment" tab set as Active 1`] = `
.c3 {
  background: #222222;
  border-right: 0;
  border: 1px solid #454545;
  color: #ffffff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1.2rem;
  font-weight: 500;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 15px;
  width: 100%;
}

.c3:last-child {
  border-right: 1px;
}

.c2 {
  background: #ffffff;
  border-right: 0;
  border: 1px solid #454545;
  color: #222222;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1.2rem;
  font-weight: 500;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 15px;
  width: 100%;
}

.c2:last-child {
  border-right: 1px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: fixed;
  width: 100%;
}

<nav
  className="c0"
>
  <Link
    className="c1"
    to="/checkout/delivery"
  >
    <span
      className="c2"
    >
      Delivery
    </span>
  </Link>
  <Link
    className="c1"
    to="/checkout/payment"
  >
    <span
      className="c3"
    >
      Payment
    </span>
  </Link>
  <Link
    className="c1"
    to="/checkout/review"
  >
    <span
      className="c2"
    >
      Review
    </span>
  </Link>
</nav>
`;

exports[`COMPONENT - Navigation renders Navigation component with "Review" tab set as Active 1`] = `
.c3 {
  background: #222222;
  border-right: 0;
  border: 1px solid #454545;
  color: #ffffff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1.2rem;
  font-weight: 500;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 15px;
  width: 100%;
}

.c3:last-child {
  border-right: 1px;
}

.c2 {
  background: #ffffff;
  border-right: 0;
  border: 1px solid #454545;
  color: #222222;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1.2rem;
  font-weight: 500;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 15px;
  width: 100%;
}

.c2:last-child {
  border-right: 1px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: fixed;
  width: 100%;
}

<nav
  className="c0"
>
  <Link
    className="c1"
    to="/checkout/delivery"
  >
    <span
      className="c2"
    >
      Delivery
    </span>
  </Link>
  <Link
    className="c1"
    to="/checkout/payment"
  >
    <span
      className="c2"
    >
      Payment
    </span>
  </Link>
  <Link
    className="c1"
    to="/checkout/review"
  >
    <span
      className="c3"
    >
      Review
    </span>
  </Link>
</nav>
`;
